### PR TITLE
Avoid pidfile race

### PR DESCRIPTION
**Pull request by Ealdwulf**
Change is_running to avoid race between readying the pid file, checking its existence, and writing to it.
@tombettany @radujipa

### CODE REVIEW

File: kano_updater/utils.py [hunk 1]
```diff
@@ -34,18 +34,56 @@
...
+    def read(self):
+        self.fh.seek(0)
+        try:
+            return int(self.fh.read(16).split("\n", 1)[0].strip())
+        except:
+            # if the pid file is corrupted, this is a big problem,
+            # but the the only chance of fixing the system is for
+            # the updater to update
+            # itself, so allow updates to go ahead.
+            return ''
```
Comment (radujipa) [inline kano_updater/utils.py]: Careful with return types here

File: kano_updater/utils.py [hunk 2]
```diff
@@ -34,18 +34,56 @@
...
+        self.fh.truncate()
+        self.fh.write(str(os.getpid()) + '\n')
+
+    def read(self):
+        self.fh.seek(0)
+        try:
+            return int(self.fh.read(16).split("\n", 1)[0].strip())
+        except:
+            # if the pid file is corrupted, this is a big problem,
+            # but the the only chance of fixing the system is for
```
Comment (radujipa) [inline kano_updater/utils.py]: "the the" typo :P

Comment (Ealdwulf) [inline kano_updater/utils.py]: Thanks, Fixed.

File: kano_updater/utils.py (hunk 1 above)
Comment (Ealdwulf) [inline kano_updater/utils.py]: Have made it None.

**Comment by radujipa** [comment]
Looks good to me :shipit:

**Comment by Ealdwulf** [comment]
Thanks,merging for testing....

**Pull request merged**